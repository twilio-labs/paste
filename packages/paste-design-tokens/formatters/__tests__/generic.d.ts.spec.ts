import { resolve } from "path";

import * as theo from "theo";

import { genericDTsTemplate } from "../generic.d.ts";

theo.registerFormat("generic.d.ts", genericDTsTemplate);

describe("generic declaration file formatter", () => {
  it("should return typescript definitions formatted tokens", async () => {
    return theo
      .convert({
        transform: {
          type: "web",
          file: resolve(__dirname, "../__fixtures__/index.yml"),
        },
        format: {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore Theo isn't typed for custom format types
          type: "generic.d.ts",
        },
      })
      .then((dTS: string) => {
        return expect(dTS).toMatchSnapshot();
      })
      .catch((error: string) => {
        // eslint-disable-next-line no-console
        console.log(`Something  went wrong: ${error}`);
        throw new Error("[dTSFormatter test]: should return typescript definitions formatted tokens threw");
      });
  });
});
