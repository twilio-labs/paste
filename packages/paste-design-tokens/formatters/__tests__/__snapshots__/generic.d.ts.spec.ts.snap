// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic declaration file formatter should return typescript definitions formatted tokens 1`] = `
"declare const Tokens: {
\\"tokens\\":{\\"background-colors\\": [
    {
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(255, 255, 255)\\";
   comment: \\"Some background\\";
   name: \\"color_background_body\\";
   altValue: \\"#FFFFFF\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(242, 47, 70)\\";
   comment: \\"Primary brand background, accessible with black text\\";
   name: \\"color_background_brand_highlight\\";
   altValue: \\"#F22F46\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(235, 15, 41)\\";
   comment: \\"Dark Primary brand background, accessible with black and white text\\";
   name: \\"color_background_brand_highlight_dark\\";
   altValue: \\"#EB0F29\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(194, 37, 55)\\";
   comment: \\"Darker Primary brand background, accessible with black and white text\\";
   name: \\"color_background_brand_highlight_darker\\";
   altValue: \\"#C22537\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(13, 18, 43)\\";
   comment: \\"Primary brand background, accessible with black text\\";
   name: \\"color_background_brand_primary\\";
   altValue: \\"#0D122B\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(246, 246, 246)\\";
   comment: \\"Some background\\";
   name: \\"color_background_default\\";
   altValue: \\"#F6F6F6\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(236, 187, 193)\\";
   comment: \\"Some background\\";
   name: \\"color_background_element_selection\\";
   altValue: \\"#ECBBC1\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(206, 80, 95)\\";
   comment: \\"Some background\\";
   name: \\"color_background_error\\";
   altValue: \\"#CE505F\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(174, 33, 49)\\";
   comment: \\"Some background\\";
   name: \\"color_background_error_dark\\";
   altValue: \\"#AE2131\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(194, 37, 55)\\";
   comment: \\"Some background\\";
   name: \\"color_background_error_light\\";
   altValue: \\"#C22537\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(230, 230, 230)\\";
   comment: \\"Some background\\";
   name: \\"color_background_row_active\\";
   altValue: \\"#E6E6E6\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(214, 214, 214)\\";
   comment: \\"Some background\\";
   name: \\"color_background_row_focus\\";
   altValue: \\"#D6D6D6\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(214, 214, 214)\\";
   comment: \\"Some background\\";
   name: \\"color_background_row_hover\\";
   altValue: \\"#D6D6D6\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(230, 230, 230)\\";
   comment: \\"Some background\\";
   name: \\"color_background_row_selected\\";
   altValue: \\"#E6E6E6\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(235, 235, 235)\\";
   comment: \\"Some background\\";
   name: \\"color_background_row_striped\\";
   altValue: \\"#EBEBEB\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(100, 209, 138)\\";
   comment: \\"Some background\\";
   name: \\"color_background_success\\";
   altValue: \\"#64D18A\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(80, 167, 110)\\";
   comment: \\"Some background\\";
   name: \\"color_background_success_dark\\";
   altValue: \\"#50A76E\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(162, 227, 184)\\";
   comment: \\"Some background\\";
   name: \\"color_background_success_light\\";
   altValue: \\"#A2E3B8\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(251, 208, 144)\\";
   comment: \\"Some background\\";
   name: \\"color_background_warning\\";
   altValue: \\"#FBD090\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(199, 141, 56)\\";
   comment: \\"Some background\\";
   name: \\"color_background_warning_dark\\";
   altValue: \\"#C78D38\\";
},
{
   type: \\"color\\";
   category: \\"background-color\\";
   value: \\"rgb(249, 177, 71)\\";
   comment: \\"Some background\\";
   name: \\"color_background_warning_light\\";
   altValue: \\"#F9B147\\";
}
],
\\"colors\\": [
    {
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(242, 47, 70)\\";
   comment: \\"Twilio brand red\\";
   name: \\"color_brand_highlight\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(235, 15, 41)\\";
   comment: \\"Twilio accessible brand red\\";
   name: \\"color_brand_highlight_dark\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(13, 18, 43)\\";
   comment: \\"Primary branding color\\";
   name: \\"color_brand_primary\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(255, 255, 255)\\";
   comment: \\"Gray Color 1\\";
   name: \\"color_gray_1\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(246, 246, 246)\\";
   comment: \\"Gray Color 2\\";
   name: \\"color_gray_2\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(235, 235, 235)\\";
   comment: \\"Gray Color 3\\";
   name: \\"color_gray_3\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(230, 230, 230)\\";
   comment: \\"Gray Color 4\\";
   name: \\"color_gray_4\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(214, 214, 214)\\";
   comment: \\"Gray Color 5\\";
   name: \\"color_gray_5\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(204, 204, 204)\\";
   comment: \\"Gray Color 6\\";
   name: \\"color_gray_6\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(153, 153, 153)\\";
   comment: \\"Gray Color 7\\";
   name: \\"color_gray_7\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(102, 102, 102)\\";
   comment: \\"Gray Color 8\\";
   name: \\"color_gray_8\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(68, 68, 68)\\";
   comment: \\"Gray Color 9\\";
   name: \\"color_gray_9\\";
   altValue: null;
},
{
   type: \\"color\\";
   category: \\"color\\";
   value: \\"rgb(34, 34, 34)\\";
   comment: \\"Gray Color 10\\";
   name: \\"color_gray_10\\";
   altValue: null;
}
],
\\"font-sizes\\": [
    {
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"0.6875rem\\";
   comment: \\"Constant typography token for font size 0\\";
   name: \\"font_size_0\\";
   altValue: \\"11px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"0.75rem\\";
   comment: \\"Constant typography token for font size 1\\";
   name: \\"font_size_1\\";
   altValue: \\"12px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"0.8125rem\\";
   comment: \\"Constant typography token for font size 2\\";
   name: \\"font_size_2\\";
   altValue: \\"13px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"0.875rem\\";
   comment: \\"Constant typography token for font size 3\\";
   name: \\"font_size_3\\";
   altValue: \\"14px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"0.9375rem\\";
   comment: \\"Constant typography token for font size 4\\";
   name: \\"font_size_4\\";
   altValue: \\"15px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"1rem\\";
   comment: \\"Constant typography token for font size 5\\";
   name: \\"font_size_5\\";
   altValue: \\"16px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"1.125rem\\";
   comment: \\"Constant typography token for font size 6\\";
   name: \\"font_size_6\\";
   altValue: \\"18px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"1.25rem\\";
   comment: \\"Constant typography token for font size 7\\";
   name: \\"font_size_7\\";
   altValue: \\"20px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"1.875rem\\";
   comment: \\"Constant typography token for font size 8\\";
   name: \\"font_size_8\\";
   altValue: \\"30px\\";
},
{
   category: \\"font-size\\";
   type: \\"font-size\\";
   value: \\"2.25rem\\";
   comment: \\"Constant typography token for font size 9\\";
   name: \\"font_size_9\\";
   altValue: \\"36px\\";
}
],
\\"spacings\\": [
    {
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"0.25rem\\";
   name: \\"space_0\\";
   altValue: \\"4px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"0.5rem\\";
   name: \\"space_1\\";
   altValue: \\"8px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"0.75rem\\";
   name: \\"space_2\\";
   altValue: \\"12px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"1rem\\";
   name: \\"space_3\\";
   altValue: \\"16px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"1.25rem\\";
   name: \\"space_4\\";
   altValue: \\"20px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"1.5rem\\";
   name: \\"space_5\\";
   altValue: \\"24px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"1.75rem\\";
   name: \\"space_6\\";
   altValue: \\"28px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"2rem\\";
   name: \\"space_7\\";
   altValue: \\"32px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"2.25rem\\";
   name: \\"space_8\\";
   altValue: \\"36px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"2.5rem\\";
   name: \\"space_9\\";
   altValue: \\"40px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"2.75rem\\";
   name: \\"space_10\\";
   altValue: \\"44px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"3rem\\";
   name: \\"space_11\\";
   altValue: \\"48px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"3.25rem\\";
   name: \\"space_12\\";
   altValue: \\"52px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"3.5rem\\";
   name: \\"space_13\\";
   altValue: \\"56px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"3.75rem\\";
   name: \\"space_14\\";
   altValue: \\"60px\\";
},
{
   category: \\"spacing\\";
   type: \\"size\\";
   value: \\"4rem\\";
   name: \\"space_15\\";
   altValue: \\"64px\\";
}
],
\\"text-colors\\": [
    {
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(242, 47, 70)\\";
   comment: \\"Twilio brand red, accessible on large text only.\\";
   name: \\"color_text_brand_highlight\\";
   altValue: \\"#F22F46\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(235, 15, 41)\\";
   comment: \\"Accessible Twilio brand red.\\";
   name: \\"color_text_brand_highlight_dark\\";
   altValue: \\"#EB0F29\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(255, 255, 255)\\";
   comment: \\"Text color used on any brand color\\";
   name: \\"color_text_brand_primary\\";
   altValue: \\"#FFFFFF\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(68, 68, 68)\\";
   comment: \\"Body text color\\";
   name: \\"color_text_default\\";
   altValue: \\"#444444\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(194, 37, 55)\\";
   comment: \\"Error text for inputs and error misc\\";
   name: \\"color_text_error\\";
   altValue: \\"#C22537\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(68, 68, 68)\\";
   comment: \\"Default icon color.\\";
   name: \\"color_text_icon_default\\";
   altValue: \\"#444444\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(255, 255, 255)\\";
   comment: \\"Inverse text color for dark backgrounds\\";
   name: \\"color_text_inverse\\";
   altValue: \\"#FFFFFF\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(235, 235, 235)\\";
   comment: \\"Weak inverse text color for dark backgrounds\\";
   name: \\"color_text_inverse_weak\\";
   altValue: \\"#EBEBEB\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(68, 68, 68)\\";
   comment: \\"Text color for field labels.\\";
   name: \\"color_text_label\\";
   altValue: \\"#444444\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(0, 112, 204)\\";
   comment: \\"Link text\\";
   name: \\"color_text_link\\";
   altValue: \\"#0070CC\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(0, 89, 163)\\";
   comment: \\"Active link text\\";
   name: \\"color_text_link_active\\";
   altValue: \\"#0059A3\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(194, 37, 55)\\";
   comment: \\"Destructive link text\\";
   name: \\"color_text_link_destructive\\";
   altValue: \\"#C22537\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(155, 29, 44)\\";
   comment: \\"Active destructive link text\\";
   name: \\"color_text_link_destructive_active\\";
   altValue: \\"#9B1D2C\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(236, 187, 193)\\";
   comment: \\"Disabled destructive link text\\";
   name: \\"color_text_link_destructive_disabled\\";
   altValue: \\"#ECBBC1\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(174, 33, 49)\\";
   comment: \\"Focus destructive link text\\";
   name: \\"color_text_link_destructive_focus\\";
   altValue: \\"#AE2131\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(174, 33, 49)\\";
   comment: \\"Hover destructive link text\\";
   name: \\"color_text_link_destructive_hover\\";
   altValue: \\"#AE2131\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(102, 169, 224)\\";
   comment: \\"Disabled link text\\";
   name: \\"color_text_link_disabled\\";
   altValue: \\"#66A9E0\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(0, 97, 192)\\";
   comment: \\"Focus link text\\";
   name: \\"color_text_link_focus\\";
   altValue: \\"#0061C0\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(0, 97, 192)\\";
   comment: \\"Hover link text\\";
   name: \\"color_text_link_hover\\";
   altValue: \\"#0061C0\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(102, 102, 102)\\";
   comment: \\"Input placeholder text.\\";
   name: \\"color_text_placeholder\\";
   altValue: \\"#666666\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(100, 209, 138)\\";
   comment: \\"Text color for success text.\\";
   name: \\"color_text_success\\";
   altValue: \\"#64D18A\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(249, 177, 71)\\";
   comment: \\"Color for texts or icons that are related to warnings on a dark background.\\";
   name: \\"color_text_warning\\";
   altValue: \\"#F9B147\\";
},
{
   type: \\"color\\";
   category: \\"text-color\\";
   value: \\"rgb(230, 230, 230)\\";
   comment: \\"Weak body text for visual hierarchy.\\";
   name: \\"color_text_weak\\";
   altValue: \\"#E6E6E6\\";
}
],
\\"z-indices\\": [
    {
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 0;
   name: \\"z-index-0\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 10;
   name: \\"z-index-10\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 20;
   name: \\"z-index-20\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 30;
   name: \\"z-index-30\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 40;
   name: \\"z-index-40\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 50;
   name: \\"z-index-50\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 60;
   name: \\"z-index-60\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 70;
   name: \\"z-index-70\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 80;
   name: \\"z-index-80\\";
   altValue: null;
},
{
   type: \\"z-index\\";
   category: \\"z-index\\";
   value: 90;
   name: \\"z-index-90\\";
   altValue: null;
}
]}
}
export default Tokens;"
`;
